pinot-broker/src/main/java/org/apache/pinot/broker/routing/BrokerRoutingManager.java

```diff
@@ -848,5 +856,15 @@ InstanceSelector.SelectionResult calculateRouting(BrokerRequest brokerRequest, l
             Collections.emptyList(), numPrunedSegments);
       }
     }
+
+    List<String> getSegments(BrokerRequest brokerRequest) {
+      Set<String> selectedSegments = _segmentSelector.select(brokerRequest);
+      if (!selectedSegments.isEmpty()) {
```

Comment:
nit: isn't this if a bit redundant?

Reply:
We want to short circuit it. This is the same as `calculateRouting()`